tests/test-api.js

```diff
@@ -0,0 +1,10 @@
+let http = require('http')
+let assert = require('./lib/assert')
+let runner = require('./lib/runner')
+const BASE_URL = 'http://localhost:8080'
+
+runner.test('base api should return 200 status code', () => {
+  http.get(BASE_URL, (res) => {
```

Comment:
uh oh, looks like we have a problem here. this test finishes asynchronously, but the tests are run synchronously. so ever though this http.get call fails (ECONNREFUSED), the test runner still thinks this test is passing `0 Failed, 1 Passed. we are doing to have to allow asynchronous testing here.

See here for some inspiration: https://mochajs.org/#asynchronous-code